Week3/solutionweek3.txt

```diff
@@ -0,0 +1,261 @@
+Write a python script to perform CRUD operation on following table of "ESM.db" database.
+
+Employee ( eid, ename, dept, basic, branch). eid must be primary key.
+
+Department : Account, Inventory, IT, HR.
+
+Peform Following Operation.
+
+1. Create table
+
+->
+from sqlite3 import *
+
+conn = connect('C:\\sqlite\\ESM.db')
+
+cur = conn.cursor()
+
+tbl = """create table Employee\
+            (\
+                eid integer primary key,\
+                ename text,\
+                dept text,\
+                basic integer,\
+                branch text\
+            );"""
+
+cur.execute(tbl)
+
+conn.commit()
+
+conn.close()
+
+2. Insert 5 Records directly, 5 records using tuple and 5 records using taking input from user
+->
+from sqlite3 import *
+
+conn = connect('C:\\sqlite\\ESM.db')
+
+cur = conn.cursor()
+
+#record which insert directly are
+"""
+1|Mitul|Account|45000|Surat
+2|rudra|IT|75000|Surat
+3|vidhi|Inventory|50000|kamrej
+4|pooja|HR|60000|Kadodra
+5|kevin|Account|3000|Bardoli
+"""
+
+#record insert using tuple
+"""
+ins = [(6,'priya','HR',50000,'Surat'),
+            (7,'krisha','IT',70000,'Vyara'),
+            (8,'Vatsal','Inventory',5000,'Vyara'),
+            (9,'dev','HR',50000,'Rajasthan'),
+            (10,'dax','Account',20000,'Baroda'),
+           ]
+
+cur.executemany("insert into employee values(?,?,?,?,?)",ins)
+"""
+
+#record insert  from taking user input
+mlis=[]
+for i in range(5):
+    tlis=[]
+    eid = int(input('enter eid :'))
+    tlis.append(eid)
+
+    ename = input('enter ename :')
+    tlis.append(ename)
+
+    dept = input('enter dept :')
+    tlis.append(dept)
+
+    basic = int(input('enter salary :'))
+    tlis.append(basic)
+
+    branch = input('enter branch :')
+    tlis.append(branch)
+
+    mlis.append(tlis)
+
+cur.executemany("insert into employee values(?,?,?,?,?)",mlis)

```

Comment:
why you are use 5 time "?"

Reply:
Because we have to insert 5 user define record.